### ISSUE TITLE
Tasks are shared across instances when using decorators

### ISSUE THREAD

**Issue by JakeCoxon** [jeffijoe/mobx-task#3]
```
class Test {
  @task test() {
    return Promise.resolve(32);
  }
}
const a = new Test();
const b = new Test();
a.test().then(() => {
  console.log(a.test.result);
  console.log(b.test.result);
})
```

This prints 32 and 32 which shows that the task is shared across instances of Test. This is undocumented but I guess I would expect each task object to be part of the instance itself.

Using the task function ie `test = task(() => ...)` works as expected so this is a viable workaround

**Comment by jeffijoe** [comment]
Thanks for the report! I'll investigate when I have some time on my hands.

**Comment by jeffijoe** [comment]
Fixed and released as 0.2.0. Thanks again!
